src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,117 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+               
+        public static bool IsInvisibleOutsideAssemblyAtSymbolLevel(ISymbol symbol)
```

Comment:
AnalyzerUtilities project has a symbol.GetResultantVisibility() method that basically does the same analysis as below. Would you be able to use that instead?